From 1f8c561494e6bbecaaa02ff9cd1b2b9216da59df Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Dan=20Hor=C3=A1k?= <dan@danny.cz>
Date: Fri, 31 Aug 2018 10:07:35 +0200
Subject: [PATCH 1/4] drop LOADLIBES variable
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Remove depreciated LOADLIBES variable from the Makefile rules, LDLIBS serves
the same purpose these days.

Signed-off-by: Dan Hor치k <dan@danny.cz>
---
 common.mak         |  2 +-
 cpuplugd/Makefile  |  2 +-
 ipl_tools/Makefile |  2 +-
 vmconvert/Makefile |  2 +-
 vmur/Makefile      |  2 +-
 ziomon/Makefile    | 10 +++++-----
 6 files changed, 10 insertions(+), 10 deletions(-)

diff --git a/common.mak b/common.mak
index 7d11b10..6029687 100644
--- a/common.mak
+++ b/common.mak
@@ -240,7 +240,7 @@ endif
 	$(CXX) $(ALL_CPPFLAGS) $(ALL_CXXFLAGS) -c $< -o $@
 
 %: %.o
-	$(LINK) $(ALL_LDFLAGS) $^ $(LOADLIBES) $(LDLIBS) -o $@
+	$(LINK) $(ALL_LDFLAGS) $^ $(LDLIBS) -o $@
 
 %.a:
 	$(AR) rcs $@ $^
diff --git a/cpuplugd/Makefile b/cpuplugd/Makefile
index a9a49ab..916638d 100644
--- a/cpuplugd/Makefile
+++ b/cpuplugd/Makefile
@@ -7,7 +7,7 @@ LDLIBS += -lm
 OBJECTS = daemon.o cpu.o info.o terms.o config.o main.o getopt.o mem.o
 
 cpuplugd: $(OBJECTS)
-	$(LINK) $(ALL_LDFLAGS) $^ $(LOADLIBES) $(LDLIBS) -o $@
+	$(LINK) $(ALL_LDFLAGS) $^ $(LDLIBS) -o $@
 
 clean:
 	rm -f cpuplugd $(OBJECTS)
diff --git a/ipl_tools/Makefile b/ipl_tools/Makefile
index 128ec3e..506d5cd 100644
--- a/ipl_tools/Makefile
+++ b/ipl_tools/Makefile
@@ -6,7 +6,7 @@ objects = main.o ccw.o fcp.o system.o shutdown.o \
 	  cmd_lsshut.o cmd_chshut.o cmd_lsreipl.o cmd_chreipl.o proc.o
 
 chreipl: $(objects)
-	$(LINK) $(ALL_LDFLAGS) $^ $(LOADLIBES) $(LDLIBS) -o $@
+	$(LINK) $(ALL_LDFLAGS) $^ $(LDLIBS) -o $@
 
 lsreipl:
 	ln -sf chreipl lsreipl
diff --git a/vmconvert/Makefile b/vmconvert/Makefile
index 4d0216c..380eb19 100644
--- a/vmconvert/Makefile
+++ b/vmconvert/Makefile
@@ -9,7 +9,7 @@ libs =  $(rootdir)/libvmdump/libvmdump.a
 objects = vmconvert.o
 
 vmconvert: $(objects) $(libs)
-	$(LINKXX) $(ALL_LDFLAGS) $^ $(LOADLIBES) $(LDLIBS) -o $@
+	$(LINKXX) $(ALL_LDFLAGS) $^ $(LDLIBS) -o $@
 
 install: all
 	$(INSTALL) -d -m 755 $(DESTDIR)$(USRBINDIR) $(DESTDIR)$(MANDIR)/man1
diff --git a/vmur/Makefile b/vmur/Makefile
index 1a6bddc..2c1c2d5 100644
--- a/vmur/Makefile
+++ b/vmur/Makefile
@@ -11,7 +11,7 @@ libs =  $(rootdir)/libvmdump/libvmdump.a \
 objects = vmur.o
 
 vmur: $(objects) $(libs)
-	$(LINKXX) $(ALL_LDFLAGS) $^ $(LOADLIBES) $(LDLIBS) -o $@
+	$(LINKXX) $(ALL_LDFLAGS) $^ $(LDLIBS) -o $@
 
 install: all
 	$(INSTALL) -d -m 755 $(DESTDIR)$(USRSBINDIR) $(DESTDIR)$(MANDIR)/man8
diff --git a/ziomon/Makefile b/ziomon/Makefile
index 778401b..61c2399 100644
--- a/ziomon/Makefile
+++ b/ziomon/Makefile
@@ -12,33 +12,33 @@ ziomon_mgr_main.o: ziomon_mgr.c
 ziomon_mgr: LDLIBS += -lm
 ziomon_mgr: ziomon_dacc.o ziomon_util.o ziomon_mgr_main.o ziomon_tools.o \
 	    ziomon_zfcpdd.o ziomon_msg_tools.o
-	$(LINK) $(ALL_LDFLAGS) $^ $(LOADLIBES) $(LDLIBS) -o $@
+	$(LINK) $(ALL_LDFLAGS) $^ $(LDLIBS) -o $@
 
 ziomon_util_main.o: ziomon_util.c ziomon_util.h
 	$(CC) -DWITH_MAIN $(ALL_CFLAGS) $(ALL_CPPFLAGS) -c $< -o $@
 ziomon_util: LDLIBS += -lm
 ziomon_util: ziomon_util_main.o ziomon_tools.o
-	$(LINK) $(ALL_LDFLAGS) $^ $(LOADLIBES) $(LDLIBS) -o $@
+	$(LINK) $(ALL_LDFLAGS) $^ $(LDLIBS) -o $@
 
 ziomon_zfcpdd_main.o: ziomon_zfcpdd.c ziomon_zfcpdd.h
 	$(CC) -DWITH_MAIN $(ALL_CFLAGS) $(ALL_CPPFLAGS) -c $< -o $@
 ziomon_zfcpdd: LDLIBS += -lm -lrt -lpthread
 ziomon_zfcpdd: ziomon_zfcpdd_main.o ziomon_tools.o
-	$(LINK) $(ALL_LDFLAGS) $^ $(LOADLIBES) $(LDLIBS) -o $@
+	$(LINK) $(ALL_LDFLAGS) $^ $(LDLIBS) -o $@
 
 ziorep_traffic: ziorep_traffic.o ziorep_framer.o ziorep_frameset.o \
 		ziorep_printers.o ziomon_dacc.o ziomon_util.o \
 		ziomon_msg_tools.o ziomon_tools.o ziomon_zfcpdd.o \
 		ziorep_cfgreader.o ziorep_collapser.o ziorep_utils.o \
 		ziorep_filters.o
-	$(LINKXX) $(ALL_LDFLAGS) $^ $(LOADLIBES) $(LDLIBS) -o $@
+	$(LINKXX) $(ALL_LDFLAGS) $^ $(LDLIBS) -o $@
 
 ziorep_utilization: ziorep_utilization.o ziorep_framer.o ziorep_frameset.o \
 		    ziorep_printers.o ziomon_dacc.o ziomon_util.o \
 		    ziomon_msg_tools.o ziomon_tools.o ziomon_zfcpdd.o \
 		    ziorep_cfgreader.o ziorep_collapser.o ziorep_utils.o \
 		    ziorep_filters.o
-	$(LINKXX) $(ALL_LDFLAGS) $^ $(LOADLIBES) $(LDLIBS) -o $@
+	$(LINKXX) $(ALL_LDFLAGS) $^ $(LDLIBS) -o $@
 
 install: all
 	$(SED) -e 's/%S390_TOOLS_VERSION%/$(S390_TOOLS_RELEASE)/' \
-- 
2.17.1


From fb5268e46eac79cfdf683ddfff6d685df6008b38 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Dan=20Hor=C3=A1k?= <dan@danny.cz>
Date: Fri, 31 Aug 2018 10:13:38 +0200
Subject: [PATCH 2/4] zkey: Drop redundant include
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Signed-off-by: Dan Hor치k <dan@danny.cz>
---
 zkey/Makefile | 1 -
 1 file changed, 1 deletion(-)

diff --git a/zkey/Makefile b/zkey/Makefile
index 68f35cf..725cb3b 100644
--- a/zkey/Makefile
+++ b/zkey/Makefile
@@ -22,7 +22,6 @@ else
 	INSTALL_TARGETS += zkey-cryptsetup-skip-cryptsetup2
 endif
 
-CPPFLAGS += -I../include
 LIBS = $(rootdir)/libutil/libutil.a
 
 detect-libcryptsetup.h:
-- 
2.17.1


From 577cbdaee51c0104465e4b77e5d931f8232d92e2 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Dan=20Hor=C3=A1k?= <dan@danny.cz>
Date: Fri, 31 Aug 2018 10:17:07 +0200
Subject: [PATCH 3/4] zkey: Be consistent when refering to libutil.a
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Signed-off-by: Dan Hor치k <dan@danny.cz>
---
 zkey/Makefile | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/zkey/Makefile b/zkey/Makefile
index 725cb3b..7e2047a 100644
--- a/zkey/Makefile
+++ b/zkey/Makefile
@@ -22,7 +22,7 @@ else
 	INSTALL_TARGETS += zkey-cryptsetup-skip-cryptsetup2
 endif
 
-LIBS = $(rootdir)/libutil/libutil.a
+libs = $(rootdir)/libutil/libutil.a
 
 detect-libcryptsetup.h:
 	echo "#include <libcryptsetup.h>" > detect-libcryptsetup.h
@@ -69,10 +69,10 @@ keystore.o: keystore.c keystore.h properties.h
 zkey-cryptsetup.o: check-dep-zkey-cryptsetup zkey-cryptsetup.c pkey.h misc.h
 
 zkey: LDLIBS = -ldl -lcrypto
-zkey: zkey.o pkey.o properties.o keystore.o $(LIBS)
+zkey: zkey.o pkey.o properties.o keystore.o $(libs)
 
 zkey-cryptsetup: LDLIBS = -ldl -lcryptsetup -ljson-c
-zkey-cryptsetup: zkey-cryptsetup.o pkey.o $(LIBS)
+zkey-cryptsetup: zkey-cryptsetup.o pkey.o $(libs)
 
 install-common:
 	$(INSTALL) -d -m 755 $(DESTDIR)$(USRBINDIR)
-- 
2.17.1


From b474b9d1a567e132f79c2a17ebbc3b56794e4797 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Dan=20Hor=C3=A1k?= <dan@danny.cz>
Date: Fri, 31 Aug 2018 04:29:39 -0400
Subject: [PATCH 4/4] zkey: Be explicit about linking the tools
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

I've met cases when the make's default rule for linking was used instead omitting
the ALL_LDFLAGS variable. The linking rule from common.mak is defined for linking
*.o files only, here we have libutil.a too.

Signed-off-by: Dan Hor치k <dan@danny.cz>
---
 zkey/Makefile | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/zkey/Makefile b/zkey/Makefile
index 7e2047a..901ddd4 100644
--- a/zkey/Makefile
+++ b/zkey/Makefile
@@ -70,9 +70,11 @@ zkey-cryptsetup.o: check-dep-zkey-cryptsetup zkey-cryptsetup.c pkey.h misc.h
 
 zkey: LDLIBS = -ldl -lcrypto
 zkey: zkey.o pkey.o properties.o keystore.o $(libs)
+	$(LINK) $(ALL_LDFLAGS) $^ $(LDLIBS) -o $@
 
 zkey-cryptsetup: LDLIBS = -ldl -lcryptsetup -ljson-c
 zkey-cryptsetup: zkey-cryptsetup.o pkey.o $(libs)
+	$(LINK) $(ALL_LDFLAGS) $^ $(LDLIBS) -o $@
 
 install-common:
 	$(INSTALL) -d -m 755 $(DESTDIR)$(USRBINDIR)
-- 
2.17.1

